ui/inspectors/system-section.reel/language-and-region.reel/language-and-region.js

```diff
@@ -46,12 +107,44 @@ exports.LanguageAndRegion = Component.specialize(/** @lends LanguageAndRegion# *
                 Promise.all(loadingPromises).then(function() {
                     this.isLoading = false;
                 });
+                var today = new Date();
+                this.dateFormatShortOptions = this._generateDateFormatConvertedList(today, this.shortDateFormats);
+                this.dateFormatMediumOptions = this._generateDateFormatConvertedList(today, this.mediumDateFormats);
+                this.dateFormatLongOptions = this._generateDateFormatConvertedList(today, this.longDateFormats);
+                this.dateFormatFullOptions = this._generateDateFormatConvertedList(today, this.fullDateFormats);
+                this.timeFormatShortOptions = this._generateDateFormatConvertedList(today, this.shortTimeFormats);
+                this.timeFormatMediumOptions = this._generateDateFormatConvertedList(today, this.mediumTimeFormats);
+                this.timeFormatLongOptions = this._generateDateFormatConvertedList(today, this.longTimeFormats);
+            }
+        }
+    },
+
+    _generateDateFormatConvertedList: {
+        value: function(today, dateOptionList) {
+            var _formattedDateList = [];
+            for (var i = 0,length = dateOptionList.length; i < length; i++) {
+                this.dateConverter.pattern = dateOptionList[i];
+                _formattedDateList.push({ label: this.dateConverter.convert(today), value: dateOptionList[i] });
             }
+            return _formattedDateList;
+        }
+    },
+
+    _checkUserSettings: {
+        value: function(user){
+            user.attributes.userSettings.timeFormatShort = user.attributes.userSettings.timeFormatShort || this.shortTimeFormats[0];
+            user.attributes.userSettings.timeFormatMedium = user.attributes.userSettings.timeFormatMedium || this.mediumTimeFormats[0];
+            user.attributes.userSettings.timeFormatLong = user.attributes.userSettings.timeFormatLong || this.longTimeFormats[0];
+            user.attributes.userSettings.dateFormatShort = user.attributes.userSettings.dateFormatShort || this.shortDateFormats[0];
+            user.attributes.userSettings.dateFormatMedium = user.attributes.userSettings.dateFormatMedium || this.mediumDateFormats[0];
+            user.attributes.userSettings.dateFormatLong = user.attributes.userSettings.dateFormatLong || this.longDateFormats[0];
+            user.attributes.userSettings.dateFormatFull = user.attributes.userSettings.dateFormatFull || this.fullDateFormats[0];
         }
     },
 
     save: {
         value: function() {
+            this.application.applicationContextService.save();
             return this.application.systemGeneralService.saveGeneralData(this.generalData);
```

Comment:
Both methods called here return a Promise, you should use Promise.all to ensure it resolves correctly.